### ISSUE TITLE
Unexpected behaviour of logit processor during beam search generation in Flax

### ISSUE THREAD

**Issue by giganttheo** [huggingface/transformers#29635]
### System Info

* `transformers` version: 4.38.2

### Who can help?

@sanchit-gandhi 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

When running beam search with a Flax model, *ie* calling `.generate` with `do_sample=False` and `num_beams>1`, the logits processors are supposed to be called with the running sequences in order to perform some processing of the logits based on the candidate sequences. However, the sequences that are passed are the ones initialized at zero, not the current `state.running_sequences`, resulting in an unexpected behaviour of the logits processors.

The line containing the error is https://github.com/huggingface/transformers/blob/b340d90738fa14bd6f81b65e4148173cbec62ff6/src/transformers/generation/flax_utils.py#L914

### Expected behavior

The current candidate sequences should be passed to the logits processor for it to perform its expected behaviour.

**Comment by amyeroberts** [comment]
Also cc @gante
